determined-ai/determined
# feat: k8s restore jobs on master restart [DET-8718]

## Description

Fault tolerance for kubernetes master by restoring jobs upon master restart.

## Test Plan

e2e tests

Manual test

Deploy a k8s cluster with helm chart option ```_reattachResources```

Then run jobs and delete the master pod and the deployment should create a new pod eventually and things should behave reasonably. Jobs shouldn't error or get deleted on master restart. 

## Commentary (optional)







## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
